design_notes/XDN11 - Error handling.md

```diff
@@ -0,0 +1,194 @@
+---
+id: XDN11
+title: Error handling
+author: mawign@microsoft.com
+status: Draft
+---
+
+# XDN11 - Error handling
+
+- Author: Manodasan Wignarajah (mawign@microsoft.com)
+- Status: Draft
+
+## Abstract
+
+This document describes the error handling story for xlang and what it will support.
+
+## Overview
+
+As described in XDN01, xlang is a system that allows a component written in any supported language to be callable from any other supported language.  To achieve this, xlang needs to support a way to translate errors from one supported language to another supported language.
+
+These translated errors may end up being handled or end up as application failures which are debugged by a developer, logged by the application, or captured in a dump.  To help facilitate this in a way that the developer has information available to them to diagnose these translated errors, xlang needs to support a way to retrieve the error context captured by a language and store it with the translated error for another language.  Errors may also end up traveling through multiple components written in different languages before reaching the failure point.  Due to this, xlang needs to support a way to store any additional error context from any language that the error propagates through.
+
+## Translation of errors
+
+Xlang's goal is to allow a component to be written using natural and familiar code for the language it is implemented in and for it to be consumed using natural and familiar code for the language which the consumer is written in.  To achieve this, xlang needs to translate errors from the natural form of one language to a natural form of that error in another language.
```

Comment:
nit, we have been using all lower case for xlang, even when it starts a sentence.